packages/cli/src/commands/deploy/index.ts

```diff
@@ -545,12 +545,20 @@ export default async (client: Client): Promise<number> => {
 
     if (!localConfig.builds || localConfig.builds.length === 0) {
       // Only add projectSettings for zero config deployments
-      createArgs.projectSettings =
-        status === 'not_linked'
-          ? {
-              sourceFilesOutsideRootDirectory,
-            }
-          : { ...localConfigurationOverrides, sourceFilesOutsideRootDirectory };
+      createArgs.projectSettings = {
+        sourceFilesOutsideRootDirectory,
+        rootDirectory:
+          sourceFilesOutsideRootDirectory && rootDirectory
+            ? rootDirectory
+            : undefined,
```

Comment:
I think I had this logic wrong in the thread. We always want to be setting the root directory:

```suggestion
        rootDirectory,
```